libcloudforensics/providers/azure/internal/account.py

```diff
@@ -0,0 +1,217 @@
+          the instance in. If none specified, then the instance will be fetched
+          from the AZ subscription.
+
+    Returns:
+      AZVirtualMachine: An Azure virtual machine object.
+
+    Raises:
+      RuntimeError: If the instance was not found in the subscription / resource
+          group.
+    """
+    instances = self.ListInstances(resource_group_name=resource_group_name)
+    if instance_name not in instances:
+      error_msg = 'Instance {0:s} was not found in subscription {1:s}'.format(
+          instance_name, self.subscription_id)
+      raise RuntimeError(error_msg)
+    return instances[instance_name]
+
+  def GetDisk(
+      self,
+      disk_name: str,
+      resource_group_name: Optional[str] = None) -> compute.AZDisk:
+    """Get disk from AZ subscription / resource group.
+
+    Args:
+      disk_name (str): The disk name.
+      resource_group_name (str): Optional. The resource group name to look
+          the disk in. If none specified, then the disk will be fetched from
+          the AZ subscription.
+
+    Returns:
+      AZDisk: An Azure Compute Disk object.
+
+    Raises:
+      RuntimeError: If the disk was not found in the subscription / resource
+          group.
+    """
+    disks = self.ListDisks(resource_group_name=resource_group_name)
+    if disk_name not in disks:
+      error_msg = 'Disk {0:s} was not found in subscription {1:s}'.format(
+          disk_name, self.subscription_id)
+      raise RuntimeError(error_msg)
+    return disks[disk_name]
+
+  def CreateDiskFromSnapshot(
+      self,
+      snapshot: compute.AZSnapshot,
+      disk_name: Optional[str] = None,
+      disk_name_prefix: Optional[str] = None,
+      disk_type: str = 'Standard_LRS') -> compute.AZDisk:
+    """Create a new disk based on a Snapshot.
+
+    Args:
+      snapshot (AZSnapshot): Snapshot to use.
+      disk_name (str): Optional. String to use as new disk name.
+      disk_name_prefix (str): Optional. String to prefix the disk name with.
+      disk_type (str): Optional. The sku name for the disk to create. Can be
+          Standard_LRS, Premium_LRS, StandardSSD_LRS, or UltraSSD_LRS. The
+          default value is Standard_LRS.
+
+    Returns:
+      AZDisk: Azure Compute Disk.
+
+    Raises:
+      RuntimeError: If the disk could not be created.
+    """
+
+    if not disk_name:
+      disk_name = common.GenerateDiskName(snapshot,
+                                          disk_name_prefix=disk_name_prefix)
+    creation_data = {
+        'location': snapshot.region,
+        'creation_data': {
+            'sourceResourceId': snapshot.resource_id,
+            'create_option': DiskCreateOption.copy
+        }
+    }
+
+    try:
+      request = self.compute_client.disks.create_or_update(
+          snapshot.resource_group_name,
+          disk_name,
+          creation_data,
+          sku=disk_type)
+      while not request.done():
+        sleep(5)  # Wait 5 seconds before checking disk status again
```

Comment:
Should we define a maximum wait time? This will loop forever otherwise

Reply:
What would be an appropriate timeout? For large disk operations things could take a little while

Reply:
Hmm in that case it might not make that much sense to have a timeout as it will be hard to predict. As long as the user can Ctrl + C and bail out of waiting, we should be good. Feel free to disregard this comment.